Library/Homebrew/utils/shell.rb

```diff
@@ -78,7 +78,7 @@ def prepend_path_in_profile(path)
       when :csh, :tcsh
         "echo 'setenv PATH #{csh_quote(path)}:$PATH' >> #{profile}"
       when :fish
-        "echo 'set -g fish_user_paths \"#{sh_quote(path)}\" $fish_user_paths' >> #{profile}"
+        "echo 'fish_add_path #{sh_quote(path)}' >> #{profile}"
```

Comment:
This function was only added in fish 3.2.0 \[1\] (also see https://github.com/Homebrew/brew/issues/8965#issuecomment-724705838). Is there a utility function we can use to get the version of the shell?

1: https://github.com/fish-shell/fish-shell/releases/tag/3.2.1

Reply:
cc @MikeMcQuaid if you have any idea 🙏

Reply:
Various shells have env vars that indicate their version \[3\]. Would it be worth adding (in another PR) a method in `utils/shell.rb` \[4\] called `preferred_version` that tries to return the user's preferred shell's version and nil if it can't be determined?

3: https://stackoverflow.com/a/38240328/553994
4: https://github.com/Homebrew/brew/blob/master/Library/Homebrew/utils/shell.rb

Reply:
I think if it works in the latest stable `brew install fish` that's sufficient.

Reply:
Latest stable fish version in brew is 3.2.2 so we should be good. https://github.com/Homebrew/homebrew-core/blob/a84689c7557c96dcf648fed27031cae6d2d3aa8f/Formula/fish.rb#L4

Reply:
Marked PR as ready for review. PTAL thanks!